lib/Re2.fx

```diff
@@ -0,0 +1,1234 @@
+/*
+    This file is a part of ficus language project.
+    See ficus/LICENSE for the licensing terms
+*/
+
+/*
+    Re2 library wrapper.
+
+    Usage: 
+    1.) Re2 library must be installed in system, see installation instructions there:
+    https://github.com/google/re2/wiki/Install
+
+    Sometimes(E.g.,, on Debian 10) it's also needed to run
+        sudo ldconfig 
+    after installation for creating appropriate soft links for shared libraries.
+
+    2.) Before compilation set environment variable FICUS_LINK_LIBRARIES to "-lre2" for providing compiler info about dependencies.
+        e.g, in bash:
+        export FICUS_LINK_LIBRARIES="-lre2"
+    3.) Use -c++ flag on compilation.
+
+
+    See https://github.com/google/re2/wiki/Syntax re2.h of Re2 source file for more detailed information about syntax and options. 
+    Wrapper function signatures have ficus-dependent specialization. They are described there.
+*/
+
+exception BadRegexp : string
+ccode 
+{
+    #include <re2/re2.h>
+    #include <vector>
+
+    int _fx_M3Re2FM14make_BadRegexpE1S(fx_str_t* arg0, fx_exn_t* fx_result); //[TODO] : Exception, declared before ccode block must be declared in program, translated to C program before this ccode for avoiding such insets. 
+
+    static int throwRe2Badregexp(const std::string& err)
+    {
+        fx_exn_t v_0 = {};
+        int fx_status = 0;
+        fx_str_t werr;
+
+        fx_status = fx_cstr2str(err.c_str(), err.size(), &werr);
+        if(fx_status >= 0) 
+        {
+            FX_CALL(_fx_M3Re2FM14make_BadRegexpE1S(&werr, &v_0), _fx_cleanup); //[TODO] : We need C-side infrastacture for user-registered exceptions. Name mangling for exception functions is a root of problem.
+            FX_THROW(&v_0, true, _fx_cleanup);
+        }
+        else
+        {
+            FX_UPDATE_BT();
+        }
+    _fx_cleanup: ;
+        fx_free_exn(&v_0);
+        return fx_status;
+    }
+
+    #define FX_RE2_THROW_BADREGEXP(str, catch_label) { fx_status = throwRe2Badregexp((str)); goto catch_label; }
+
+    void fx_re2_free(void* ptr)
+    {
+        using namespace re2;
+        delete static_cast<RE2*>(ptr);
+    }
+
+    static void free_sub_args(std::vector<re2::RE2::Arg*>& a_sub_args)
+    {
+        for(size_t arg_num = 0; arg_num < a_sub_args.size(); arg_num++)
+        {
+            delete a_sub_args[arg_num];
+        }
+    }
+
+    static int make_sub_args(std::vector<re2::StringPiece>& a_sub_string_pieces, std::vector<re2::RE2::Arg*>& a_sub_args)
```

Comment:
can we convert `sub_args` into `std::vector<re2::RE2::Arg>` instead of `std::vector<re2::RE2::Arg*>`? And then we can get simplify `make_sub_args()`, get rid of `free_sub_args` and probably simplify some other code too.

Reply:
Done with adding another vector for keeping Args theirself.